Front-end/src/MatchMentor.jsx

```diff
@@ -0,0 +1,244 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+
+import "bootstrap/dist/css/bootstrap.min.css";
+import "semantic-ui-css/semantic.min.css";
+
+import NavBar from "./components/Navbar";
+import {
+  Form,
+  FormField,
+  FormSelect,
+  FormInput,
+  CardMeta,
+  CardHeader,
+  CardGroup,
+  CardDescription,
+  CardContent,
+  Button,
+  Card,
+  Image,
+} from "semantic-ui-react";
+
+const MatchMentor = (props) => {
+  const [date, setDate] = useState("");
+  const [user, setUser] = useState();
+  const [info, setInfo] = useState("");
+  const [profile, setProfile] = useState();
+  const [mentors, setMentors] = useState([]);
+  const [func, setFunc] = useState(false);
+  const invalid = "Invalid Token";
+
+  async function auth() {
+    let token = localStorage.getItem("accessToken");
+    await fetch("http://localhost:4000/auth", {
+      method: "GET",
+      headers: {
+        "Content-Type": "Application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((data) =>
+      data.json().then((data) => {
+        if (data === invalid) {
+          token = localStorage.getItem("refreshToken");
+          fetch("http://localhost:4000/token", {
+            method: "POST",
+            headers: {
+              "Content-Type": "Application/json",
+            },
+            body: JSON.stringify({ token: token }),
+          }).then((data) =>
+            data.json().then((data) => {
+              if (data === invalid) {
+                window.location.href = "/";
+              } else {
+                localStorage.setItem("accessToken", data);
+              }
+            })
+          );
+        } else {
+          setUser(data);
+        }
+      })
+    );
+  }
+  async function logOut() {
+    const token = localStorage.getItem("accessToken");
+    await fetch("http://localhost:4000/logout", {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "Application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((data) =>
+      data.json().then((data) => {
+        localStorage.removeItem("accessToken");
+        localStorage.removeItem("refreshToken");
+        window.location.href = "/";
+      })
+    );
+  }
+  async function member2() {
```

Comment:
Give it a better name? member2 is not evident on what it meant

Reply:
Updated